### ISSUE TITLE
Tiling windows sized incorrectly

### ISSUE THREAD

**Issue by Baggykiin** [baskerville/bspwm#575]
For some reason, bspwm seems to set the size of any tiling windows on my second monitor incorrectly, making them _way_ too wide (65522×1043 pixels), which causes the majority of the window to fall outside the bounds of my monitor. Many applications also crash or behave incorrectly because they can't handle a window that large.

The issue only seems to occur on tiling windows. Fullscreen windows will fill the monitor as expected and do not go out of bounds in any direction, and floating windows cannot be dragged off the monitor (except for to the left, onto my main monitor, of course).

If I rearrange the monitors so my second monitor is placed to the left of my first monitor, the tiling windows from the second monitor still go out of bounds on the right side, also obscuring the windows on my main monitor.

Xrandr seems to report the correct resolution:
```
Screen 0: minimum 8 x 8, current 3840 x 1080, maximum 32767 x 32767
eDP1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 350mm x 190mm
   1920x1080     60.02*+
   1400x1050     59.98  
   1600x900      60.00  
   1280x1024     60.02  
   1280x960      60.00  
   1368x768      60.00  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   640x360       60.00  
HDMI1 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 600mm x 340mm
   2560x1440     59.95 +
   1920x1200     59.95  
   1920x1080     60.00*   50.00    50.00    59.94  
   1920x1080i    60.00    50.00    59.94  
   1600x1200     60.00  
   1680x1050     59.88  
   1280x1024     75.02    60.02  
   1440x900      74.98    59.90  
   1280x720      60.00    50.00    59.94  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   720x576       50.00  
   720x576i      50.00  
   720x480       60.00    59.94  
   720x480i      60.00    59.94  
   640x480       75.00    72.81    66.67    60.00    59.94  
   720x400       70.08  
```

The monitor has a native resolution of 2560x1440, and the issue seems to occur no matter what resolution I set it to. I currently have it set to 1920x1080 to match the resolution of my laptop, for convenience.

Running xwininfo on a tiling window going offscreen returns the following:

```
xwininfo: Window id: 0x800009 "urxvt"

  Absolute upper-left X:  1925
  Absolute upper-left Y:  28
  Relative upper-left X:  1925
  Relative upper-left Y:  28
  Width: 65522
  Height: 1043
  Depth: 24
  Visual: 0x20
  Visual Class: TrueColor
  Border width: 2
  Class: InputOutput
  Colormap: 0x22 (installed)
  Bit Gravity State: ForgetGravity
  Window Gravity State: NorthWestGravity
  Backing Store State: NotUseful
  Save Under State: no
  Map State: IsViewable
  Override Redirect State: no
  Corners:  +1925+28  --63611+28  --63611-5  +1925-5
  -geometry 9358x74+1925-5
```

Edit: `bspc query -T -m 1` output:
```
{
    "name": "1",
    "id": 4194307,
    "randrId": 68,
    "wired": true,
    "stickyCount": 0,
    "windowGap": 5,
    "borderWidth": 2,
    "focusedDesktopId": 4194309,
    "padding": {
        "top": 23,
        "right": 1920,
        "bottom": 0,
        "left": 0
    },
    "rectangle": {
        "x": 1920,
        "y": 0,
        "width": 1920,
        "height": 1080
    },
    "desktops": [
        {
            "name": "I",
            "id": 4194309,
            "layout": "tiled",
            "windowGap": 5,
            "borderWidth": 2,
            "focusedNodeId": 14680073,
            "padding": {
                "top": 0,
                "right": 0,
                "bottom": 0,
                "left": 0
            },
            "root": {
                "id": 14680073,
                "splitType": "vertical",
                "splitRatio": 0.5,
                "birthRotation": 0,
                "vacant": false,
                "hidden": false,
                "sticky": false,
                "private": false,
                "locked": false,
                "presel": null,
                "rectangle": {
                    "x": 1925,
                    "y": 28,
                    "width": 65531,
                    "height": 1052
                },
                "firstChild": null,
                "secondChild": null,
                "client": {
                    "className": "URxvt",
                    "instanceName": "urxvt",
                    "borderWidth": 2,
                    "state": "tiled",
                    "lastState": "tiled",
                    "layer": "normal",
                    "lastLayer": "normal",
                    "urgent": false,
                    "shown": true,
                    "tiledRectangle": {
                        "x": 1925,
                        "y": 28,
                        "width": 65522,
                        "height": 1043
                    },
                    "floatingRectangle": {
                        "x": 2592,
                        "y": 368,
                        "width": 571,
                        "height": 340
                    }
                }
            }
        }
    ]
}

```

**Comment by baskerville** [comment]
Is `honor_size_hints` set?

**Comment by Baggykiin** [comment]
It was set to `false`, and the issue seems to occur regardless of its value.

After some more testing, I've figured out that `lemonbar` seems to be triggering the problem. The desktop behaves correctly until lemonbar is started. It doesn't seem to depend on any specific configuration; simply running `lemonbar -p` is enough to cause it. Same goes for when it's started as `lemonbar -p -g 10x10+10+10` (which runs lemonbar in a 10x10 window at x=10;y=10).

**Comment by neeasade** [comment]
out of curiosity, does launching lemonbar with the `-d` flag change the behavior? (would eliminate strut handling involvement)

**Comment by Baggykiin** [comment]
That does indeed seem to fix the issue for me.

**Comment by baskerville** [comment]
Is this still happening after 89adecf?

**Comment by Baggykiin** [comment]
It is. Up to this point I've tested everything on that commit.

**Comment by baskerville** [comment]
It should be fixed by 201f6e6.

**Comment by Baggykiin** [comment]
Yup, it works without the `-d` flag now.
